### PR TITLE
make brush tool consistent with other tools; add dither

### PR DESCRIPTION
**Pull request by microlith57**
currently, the brush tool directly modifies the level while the mouse button is down, which is not consistent with the other tools (which only modify the level when the button is released); this fixes the inconsistency.

additionally, i made a dither tool that behaves the same (or at least very similarly) to ahorn's.

### CODE REVIEW

File: src/tools/brush.lua [hunk 1]
```diff
@@ -43,18 +59,87 @@ function tool.mouseclicked(x, y, button, istouch, pressed)
...
+local function addPoint(x, y, roomSize)
+    if not roomSize then
+        local room = state.getSelectedRoom()
+        local tilesMatrix = room[tool.layer].matrix
+        local roomHeight
+        roomSize = tilesMatrix:size()
+    end
+    local roomWidth, roomHeight = roomSize
+
+    local index = x + y * roomWidth
```
Comment (microlith57) [inline src/tools/brush.lua:100]: is this ok for points outside of the room?

File: src/lang/en_gb.lang [hunk 2]
```diff
@@ -903,7 +910,7 @@ entities.flingBirdIntro.attributes.description.crashes=Determines which cutscene
 entities.lightningBlock.placements.name.breaker_box=Lightning Breaker Box
 entities.lightningBlock.attributes.description.music_session=Whether the music should be stored in session when box is broken.
 entities.lightningBlock.attributes.description.flipX=Whether the sprite should be vertically mirrored.
-entities.lightningBlock.attributes.description.flag=Whether the breaker box should disable all lightning entities without the perLevel attribute set when loading a new room. 
+entities.lightningBlock.attributes.description.flag=Whether the breaker box should disable all lightning entities without the perLevel attribute set when loading a new room.
```
Comment (microlith57) [inline src/lang/en_gb.lang:913]: these changes were done by my ide

File: src/tools/brush.lua [hunk 3]
```diff
@@ -43,18 +59,87 @@ function tool.mouseclicked(x, y, button, istouch, pressed)
...
+        elseif tool.mode == "dither" then
+            local room = state.getSelectedRoom()
+
+            if room then
+                local px, py = viewportHandler.getRoomCoordindates(room, x, y)
+                local tx, ty = viewportHandler.pixelToTileCoordinates(px, py)
+
+                brushToolUtils.startTileSnapshot(tool)
+                brushHelper.placeTile(room, tx + 1, ty + 1, tool.material, tool.layer)
+                brushHelper.placeTile(room, tx + 2, ty + 2, tool.material, tool.layer)
```
Comment (microlith57) [inline src/tools/brush.lua:73]: when i tested this, the second tile was placed but didn't display until re-autotiled; i have observed this with other tools so i think it's a display bug

File: src/tools/brush.lua [hunk 4]
```diff
@@ -43,18 +59,87 @@ function tool.mouseclicked(x, y, button, istouch, pressed)
...
+                brushToolUtils.stopTileSnapshot(tool)
+
+                tool.lastTileX, tool.lastTileY = tx + 1, ty + 1
+                tool.lastMaterial = tool.material
+
+                tool.lastX, tool.lastY = x, y
+            end
+        elseif tool.mode == "loenn" then
+            -- TODO
+            return {}
```
Comment (microlith57) [inline src/tools/brush.lua:84]: left this return in by mistake, but it has no effect

**Review by microlith57 — COMMENTED** [review]
i'd suggest making the lönn brush stamp before merging this

**Review by Cruor — COMMENTED** [review]
Targeting wrong branch, has no UI changes.
I don't agree with the solution for adding multiple brushes, it is not modular at all. This PR is also too broad in my opinion, I don't see why it should be changing brushing behaviour.
There are already known issues with how some brushes behave (previously fixed some cases of it) and I'd rather fix something like this at that point.

**Comment by microlith57** [comment]
seems reasonable

**Pull request closed without merging**